uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/components/leave-calendar.component.spec.js

```diff
@@ -216,6 +243,106 @@
           });
         });
 
+        describe('filter by assignee', function () {
+          it('does *not* have such a filter for staff', function () {
+            expect(controller.filtersByAssignee).not.toBeDefined();
+          });
+
+          describe('when user is Admin', function () {
+            beforeEach(function () {
+              currentContact.role = 'admin';
+
+              compileComponent();
+            });
+
+            it('has the filter available', function () {
+              expect(controller.filtersByAssignee).toBeDefined();
+            });
+
+            describe('when filter by assignee is set to "Me"', function () {
+              beforeEach(function () {
+                selectFilterByAssignee('me');
+              });
+
+              it('does *not* load contracts', function () {
+                expect(Contract.all).not.toHaveBeenCalledWith();
+              });
+
+              it('loads only managees', function () {
+                expect(Contact.all).not.toHaveBeenCalledWith();
+                expect(Contact.leaveManagees).toHaveBeenCalledWith(currentContact.id);
+              });
+            });
+
+            describe('when filter by assignee is set to "Unassigned"', function () {
+              beforeEach(function () {
+                selectFilterByAssignee('unassigned');
+              });
+
+              it('loads all contracts', function () {
+                expect(Contract.all).toHaveBeenCalledWith();
+              });
+
+              it('loads all contacts', function () {
+                expect(Contact.leaveManagees).toHaveBeenCalledWith(undefined, {
+                  unassigned: true
+                });
+              });
+            });
+
+            describe('when filter by assignee is set to "All"', function () {
+              beforeEach(function () {
+                selectFilterByAssignee('all');
+              });
+
+              it('loads all contracts', function () {
+                expect(Contract.all).toHaveBeenCalledWith();
+              });
+
+              it('loads all contacts', function () {
+                expect(Contact.all).toHaveBeenCalledWith();
+              });
+            });
+          });
+
+          afterEach(function () {
+            currentContact.role = 'staff';
+          });
+        });
+
+        describe('hint for admin filtering logic', function () {
```

Comment:
if a hint component is created there wouldn't be a need to test hint logic each time you use it

Reply:
We decided to leave the current implementation.